Description: Use reqwest requester to replace deprecated hyper requester
Applied-Upstream: https://gitlab.com/schmiddi-on-mobile/railway-backend/-/merge_requests/22
Author: Blair Noctis <ncts@debian.org>
Last-Update: 2025-04-01
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -38,2 +38,3 @@
 package = "railway-core"
+features = ["reqwest-requester"]
 
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -11,3 +11,3 @@
     Journey, JourneysOptions, Location, Mode, ProductsSelection, Provider, Requester,
-    RequesterBuilder,
+    ReqwestRequesterBuilder, RequesterBuilder,
 };
@@ -199,3 +199,3 @@
     use rcore::{
-        HyperRustlsRequesterBuilder, JourneysOptions, Location, LocationsOptions, Place, Station,
+        JourneysOptions, Location, LocationsOptions, Place, Station,
     };
@@ -208,3 +208,3 @@
     ) -> Result<(), Box<dyn std::error::Error>> {
-        let client = SearchChClient::new(HyperRustlsRequesterBuilder::default());
+        let client = SearchChClient::new(ReqwestRequesterBuilder::default());
         let locations = client
@@ -236,3 +236,3 @@
     ) -> Result<(), Box<dyn std::error::Error>> {
-        let client = SearchChClient::new(HyperRustlsRequesterBuilder::default());
+        let client = SearchChClient::new(ReqwestRequesterBuilder::default());
         let journeys = client
